homeassistant/components/zwave_js/climate.py

```diff
@@ -0,0 +1,332 @@
+    ThermostatMode,
+)
+from zwave_js_server.model.value import Value as ZwaveValue
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    ATTR_HVAC_MODE,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_FAN,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    DOMAIN as CLIMATE_DOMAIN,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_TARGET_TEMPERATURE_RANGE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DATA_CLIENT, DATA_UNSUBSCRIBE, DOMAIN
+from .discovery import ZwaveDiscoveryInfo
+from .entity import ZWaveBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+# Map Z-Wave HVAC Mode to Home Assistant value
+# Note: We treat "auto" as "heat_cool" as most Z-Wave devices
+# report auto_changeover as auto without schedule support.
+ZW_HVAC_MODE_MAP: Dict[int, str] = {
+    ThermostatMode.OFF: HVAC_MODE_OFF,
+    ThermostatMode.HEAT: HVAC_MODE_HEAT,
+    ThermostatMode.COOL: HVAC_MODE_COOL,
+    # Z-Wave auto mode is actually heat/cool in the hass world
+    ThermostatMode.AUTO: HVAC_MODE_HEAT_COOL,
+    ThermostatMode.AUXILIARY: HVAC_MODE_HEAT,
+    ThermostatMode.FAN: HVAC_MODE_FAN_ONLY,
+    ThermostatMode.FURNANCE: HVAC_MODE_HEAT,
+    ThermostatMode.DRY: HVAC_MODE_DRY,
+    ThermostatMode.AUTO_CHANGE_OVER: HVAC_MODE_HEAT_COOL,
+    ThermostatMode.HEATING_ECON: HVAC_MODE_HEAT,
+    ThermostatMode.COOLING_ECON: HVAC_MODE_COOL,
+    ThermostatMode.AWAY: HVAC_MODE_HEAT_COOL,
+    ThermostatMode.FULL_POWER: HVAC_MODE_HEAT,
+}
+
+ZWAVE_SETPOINT_PROPERTY = "setpoint"
+
+
+class OperatingMode(IntEnum):
+    """Enum with all (known/used) Z-Wave OperatingModes."""
+
+    # https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/commandclass/ThermostatOperatingStateCC.ts#L38-L51
+    IDLE = 0
+    HEATING = 1
+    COOLING = 2
+    FAN_ONLY = 3
+    PENDING_HEAT = 4
+    PENDING_COOL = 5
+    VENT_ECONOMIZER = 6
+    AUX_HEATING = 7
+    SECOND_STAGE_HEATING = 8
+    SECOND_STAGE_COOLING = 9
+    SECOND_STAGE_AUX_HEAT = 10
+    THIRD_STAGE_AUX_HEAT = 11
+
+
+HVAC_CURRENT_MAP: Dict[int, str] = {
+    OperatingMode.IDLE: CURRENT_HVAC_IDLE,
+    OperatingMode.PENDING_HEAT: CURRENT_HVAC_IDLE,
+    OperatingMode.HEATING: CURRENT_HVAC_HEAT,
+    OperatingMode.PENDING_COOL: CURRENT_HVAC_IDLE,
+    OperatingMode.COOLING: CURRENT_HVAC_COOL,
+    OperatingMode.FAN_ONLY: CURRENT_HVAC_FAN,
+    OperatingMode.VENT_ECONOMIZER: CURRENT_HVAC_FAN,
+    # not defined in ozw integration but we will provide them anyway
```

Comment:
No, I added that because as I mentioned in the PR, I modeled this off of OZW, but the mappings that follow the comment were not there in the OZW integration

Reply:
If we need a comment, we should write the comment without taking ozw into consideration. That's a different integration and not relevant for the code, besides as context when reviewing this PR.

Reply:
OK will remove since it only matters in the context of OZW 👍🏾